model/llm/llm.go

```diff
@@ -145,29 +163,21 @@ func (m *Model) generateTestsForFile(ctx task.Context) (assessment metrics.Asses
 		retry.DelayType(retry.BackOffDelay),
 		retry.LastErrorOnly(true),
 		retry.OnRetry(func(n uint, err error) {
-			ctx.Logger.Printf("Attempt %d/%d: %s", n+1, m.queryAttempts, err)
+			log.Printf("Attempt %d/%d: %s", n+1, m.queryAttempts, err)
 		}),
 	); err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
-	assessment, testContent, err := prompt.ParseResponse(response)
+	assessment, testContent, err = prompt.ParseResponse(response)
```

Comment:
Sorry for noticing now, would extract the parsing logic from the query helper. Cause that parsing might not always be what we want. The query model should really only be the query and return the response.

Reply:
And while we are at it, just call the function `query`. Otherwise it is `Model.queryModel` which is doubled.

Reply:
function not renamed yet :disappointed: